homeassistant/components/kem/sensor.py

```diff
@@ -0,0 +1,219 @@
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    KemSensorEntityDescription(
+        key="lubeOilTempF",
+        translation_key="lube_oil_temperature",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    KemSensorEntityDescription(
+        key="controllerTempF",
+        translation_key="controller_temperature",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    KemSensorEntityDescription(
+        key="engineCompartmentTempF",
+        translation_key="engine_compartment_temperature",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    KemSensorEntityDescription(
+        key="engineFrequencyHz",
+        translation_key="engine_frequency",
+        native_unit_of_measurement=UnitOfFrequency.HERTZ,
+        device_class=SensorDeviceClass.FREQUENCY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    KemSensorEntityDescription(
+        key="status",
+        icon="mdi:home-lightning-bolt",
+        translation_key="generator_status",
+        use_device_key=True,
+    ),
+    KemSensorEntityDescription(
+        key="engineState",
+        icon="mdi:home-lightning-bolt",
+        translation_key="engine_state",
+    ),
+    KemSensorEntityDescription(
+        key="powerSource",
+        icon="mdi:home-lightning-bolt",
+        translation_key="power_source",
+    ),
+    KemSensorEntityDescription(
+        key="totalOperationHours",
+        translation_key="total_operation",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    KemSensorEntityDescription(
+        key="totalRuntimeHours",
+        translation_key="total_runtime",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        use_device_key=True,
+    ),
+    KemSensorEntityDescription(
+        key="runtimeSinceLastMaintenanceHours",
+        translation_key="runtime_since_last_maintenance",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.HOURS,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
```

Comment:
Other suggestion, you could look into making this a timestamp sensor so the state of the entity is just the last moment maintenance happened, this way the state is always up to date

Reply:
This data is the number of hours the generator has ran since the last maintenance. Typically the generator is not running - except during a power outage or a weekly exercise. Kohler recommends checking the engine oil after 20 hours of run-time. So what you do is set an alert on this sensor when it exceeds 20 hours; check the oil, and select on  the panel to update the last maintenance.

Reply:
Oh right, I made the assumption that it would update 1 hour every hour, in which case a timestamp entity would make more sense